### PR TITLE
[Merged by Bors] - feat: `Nat.cast_nonpos`

### PR DESCRIPTION
**Pull request by YaelDillies**
For simp confluence

From GrowthInGroups

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Review by Ruben-VandeVelde — APPROVED** [review]
I suppose

maintainer merge

File: Mathlib/Data/Nat/Cast/Order/Basic.lean [hunk 1]
```diff
@@ -98,6 +98,8 @@ theorem cast_lt_one : (n : α) < 1 ↔ n = 0 := by
 @[simp, norm_cast]
 theorem cast_le_one : (n : α) ≤ 1 ↔ n ≤ 1 := by rw [← cast_one, cast_le]
 
+@[simp] lemma cast_nonpos : (n : α) ≤ 0 ↔ n ≤ 0 := by norm_cast
```
Comment (urkud) [inline Mathlib/Data/Nat/Cast/Order/Basic.lean:101]: Should we simplify to `n ≤ 0` or `n = 0`?

Comment (j-loreaux) [reply Mathlib/Data/Nat/Cast/Order/Basic.lean:101]: I suggest `n = 0`, especially as that agrees with `cast_lt_one` above.

**Review by j-loreaux — COMMENTED** [review]
bors merge

**Pull request closed without merging**